### PR TITLE
[YUNIKORN-2861] - Angular upgrade to v18

### PR DESCRIPTION
**Pull request by dcoric**
### What is this PR for?
This PR updates the Angular version to the v18. I have done it in 2 commits - to v17 and to v18. Small changes were required with the zone.js framework import. 
The only condition is - a requirement for the minimum node version: v18.19.x 

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
[YUNIKORN-2861 Angular upgrade to v18](https://issues.apache.org/jira/browse/YUNIKORN/YUNIKORN-2861)

### How should this be tested?
I have executed the test suite and application build. Everything appears to be functional and correct.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-18.20
+v18.20.4
```
Comment (chenyulin0719) [inline .nvmrc:1]: Building the web image will fail. 
> REGISTRY=apache VERSION=latest make clean image

The reason is the unexpected 'v' in the resolved node tag name in Dockerfile.
Error Message:
`ERROR: failed to solve: node:v18.20.4-alpine: docker.io/library/node:v18.20.4-alpine: not found`

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
Built web image failed.  Could you also change this PR title to the Jira name? So this PR could be linked to the corresponding Jira automatically.

File: .nvmrc (hunk 1 above)
Comment (dcoric) [reply .nvmrc]: Thanks. I had to regenerate `pnpm-lock.yaml` as well.

**Comment by dcoric** [comment]
Additional note: If the [module federation PR](https://github.com/apache/yunikorn-web/pull/198) goes through before this one, `@angular-architects/module-federation` dependency needs to be updated to the `18.x` version that matches the rest of the dependency structure.

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
@dcoric The PR of [module federation](https://github.com/apache/yunikorn-web/pull/198) is merged.  Rebased required, thanks!

**Comment by dcoric** [comment]
> @dcoric The PR of [module federation](https://github.com/apache/yunikorn-web/pull/198) is merged. Rebased required, thanks!

Thanks. I have rebased and updated minor versions. The allocations drawer test reported that `MatDrawer` had to be removed from declarations in `allocations-drawer.component.spec.ts`. Doing that fixed the failing test. 

I have tested the changes by running the application in localhost, `make build` and `make test_js`

**Review by chenyulin0719 — APPROVED** [review]
LGTM, checked the web after running the commands below:
- make install
- make test
-  make clean image

Also checked the web container running as expected.

**Pull request closed without merging**